subsys/bluetooth/host/id.c

```diff
@@ -286,6 +286,7 @@ static void le_rpa_invalidate(void)
 			return;
 		}
 		bool rpa_expired_data[bt_dev.id_count];
+		memset(rpa_expired_data, true, sizeof(rpa_expired_data));
```

Comment:
Hmm, if it's always initialized to `true` and never modified, why does the check even exist in `adv_rpa_clear_data`? :o:

Reply:
Because on `adv_rpa_invalidate` it can be set to false.

Reply:
Basically, the default behavior is to invalidate it, unless the application prevents it.

Reply:
Ah, you are correct. Thanks :) Then this looks correct to me since `adv_rpa_invalidate` may only set it to `false`, so it should be initialized to `true`